service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ApplicantController.java

```diff
@@ -24,33 +31,70 @@
 
     @Autowired
     private MapperService mapperService;
+    private final ApplicantMigrationService applicantMigrationService;
+
+    @Autowired
+    public ApplicantController(MapperService mapperService,
+                               ApplicantMigrationService applicantMigrationService) {
+        this.mapperService = mapperService;
+        this.applicantMigrationService = applicantMigrationService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+
+        return applicantMigrationService.setMigratedValue(caseDetails);
+    }
 
     @SuppressWarnings("unchecked")
     @PostMapping("/mid-event")
     public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        ImmutableList.Builder<String> validationErrors = ImmutableList.builder();
 
-        Map<String, Object> applicantData = (Map<String, Object>) caseDetails.getData().get("applicant");
-        Applicant applicant = mapperService.mapObject(applicantData, Applicant.class);
-        List<String> validationErrors = new ArrayList<String>();
+        if (caseDetails.getData().containsKey("applicants")) {
+            List<Map<String, Object>> migratedApplicantsObject =
+                (List<Map<String, Object>>) caseDetails.getData().get("applicants");
+
+            migratedApplicantsObject.stream()
+                .map(applicant ->
+                    mapperService.mapObject((Map<String, Object>) applicant.get("value"), Applicant.class))
+                .map(Applicant::getParty)
+                .map(ApplicantParty::getPbaNumber)
+                .filter(Objects::nonNull)
+                .forEach(pbaNumber -> {
+                    String newPbaNumberData = PBANumberHelper.updatePBANumber(pbaNumber);
+                    validationErrors.addAll(validatePBANumber(newPbaNumberData));
+                });
+
+            caseDetails.getData().put("applicants", migratedApplicantsObject);
 
-        if (applicant.getPbaNumber() == null || applicant.getPbaNumber().isBlank()) {
-            return AboutToStartOrSubmitCallbackResponse.builder()
-                .data(caseDetails.getData())
-                .errors(validationErrors)
-                .build();
         } else {
+            Map<String, Object> applicantData = (Map<String, Object>)
+                defaultIfNull(caseDetails.getData().get("applicant"), null);
+
+            OldApplicant applicant = mapperService.mapObject(applicantData, OldApplicant.class);
```

Comment:
how does this line deal with mapping null when the above case defaults to null?

Reply:
commenting for the comment above but i think we need to check in the mid event on the new applicants structure that the pba stuff is still working?